File: html/.work/probe.py
import sys,time,threading,functools,http.server,socketserver,os,json
from playwright.sync_api import sync_playwright
ROOT=os.path.abspath('..'); PORT=8766
class Q(http.server.SimpleHTTPRequestHandler):
    def log_message(self,*a): pass
socketserver.TCPServer.allow_reuse_address=True
srv=socketserver.ThreadingTCPServer(("127.0.0.1",PORT),functools.partial(Q,directory=ROOT)); threading.Thread(target=srv.serve_forever,daemon=True).start()
W=int(sys.argv[1]); js=sys.argv[2]
with sync_playwright() as p:
    b=p.chromium.launch(); pg=b.new_page(viewport={"width":W,"height":900})
    pg.add_init_script("window.setInterval=function(){return 0}")
    pg.goto(f"http://127.0.0.1:{PORT}/index.html",wait_until="load"); time.sleep(1.5)
    print(pg.evaluate(js))
    b.close()
srv.shutdown()

File: html/.work/p2/probe.mjs
import { chromium } from 'playwright';
const b = await chromium.launch();
const p = await b.newPage({ viewport:{width:1440,height:1000} });
await p.goto('https://ichess.edu.vn/cua-hang/', {waitUntil:'load'});
await p.waitForTimeout(4000);
console.log(await p.evaluate(() => {
  const ul = document.querySelector('ul.product-grid');
  const li = ul.children[0];
  return JSON.stringify({ n: ul.children.length, tag: li && li.tagName,
    d: li && getComputedStyle(li).display, v: li && getComputedStyle(li).visibility,
    rect: li && JSON.stringify(li.getBoundingClientRect()) });
}));
await b.close();
